src/workloads/execution/TimeSeriesRangeDelete.yml

```
@@ -0,0 +1,154 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/server-execution"
+Description: |
+  This test establishes a baseline for time-based range deletes on time-series collections.
+
+  We insert 1000 independent series with 100 buckets in each series, and each bucket has 100
+  documents.
+
+  The buckets have meta fields with values from 0 to 999, each of which has 100 measurements with
+  timestamps within an hour with 36-second intervals. For example, in a bucket, the measurements
+  look like:
+  {t: 2023-01-01T5:00:00, m: 42, ...},
+  {t: 2023-01-01T5:00:36, m: 42, ...},
+  ...
+  {t: 2023-01-01T5:59:24, m: 42, ...}
+
+  Then we delete data that spans across three buckets for each series, causing two partial-bucket
+  deletions and one full-bucket deletion. This tests the use case of data correction where the
+  application deletes data within some time ranges.
+
+GlobalDefaults:
+  dbname: &db test
+  coll: &coll Collection0
+  nSensors: &nSensors 1000
+  nBucketsPerSensor: &nBucketsPerSensor 100
+  nMeasurementsPerBucket: &nMeasurementsPerBucket 100
+  nop: &Nop {Nop: true}
+  MaxPhases: &MaxPhases 4
+
+Actors:
+- Name: CreateTimeSeriesCollection
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *db
+        Operation:
+          OperationName: RunCommand
+          OperationCommand:
+            # Each bucket holds data within one hour time span.
+            {
+              create: *coll,
+              timeseries: {timeField: "t", metaField: "sensorId", granularity: "seconds"}
+            }
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *db
+        Collection: *coll
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount:
+          ^NumExpr:
+            withExpression: "nSensors * nBucketsPerSensor * nMeasurementsPerBucket"
+            andValues: {
+              nSensors: *nSensors,
+              nBucketsPerSensor: *nBucketsPerSensor,
+              nMeasurementsPerBucket: *nMeasurementsPerBucket
```

[optional] Following comment or something similar along the line.
```suggestion
              # 1hour time span per a bucket (== 3600 seconds) / 36000 millis (== 36 seconds) increment per each measurement == 100 measurements per a bucket
              nMeasurementsPerBucket: *nMeasurementsPerBucket
```

---

done.